### DISCUSSION TITLE
Invalid prop 'component' supplied to 'Route'

### DISCUSSION THREAD

**Discussion by GonzalesMatthew** [nss-evening-cohort-14/evening-client-side#258] [Help Me!] [answered]
Matthew Gonzales

I don't see any related ticket on this but...

I'm trying to understand why I am getting this pesky error. It doesn't seem to break anything, but I don't know what to do to make it go away.
![Screen Shot 2021-05-27 at 9 11 49 PM](https://user-images.githubusercontent.com/16343050/119919492-2f6dea00-bf30-11eb-9939-2ca47914f5bd.png)

I believe it is referencing this part of my Private Route logic , but I still don't know how to fix it.

![Screen Shot 2021-05-27 at 9 13 15 PM](https://user-images.githubusercontent.com/16343050/119919599-60e6b580-bf30-11eb-9eaf-00d81c7cce49.png)

This code is found here and it matches to how we've been using Private Routes with the exception that I am using 'admin' here instead of 'user':
https://github.com/GonzalesMatthew/react-personal-bio-site/blob/main/src/helpers/Routes.js

I googled the error and someone mentioned that my export could be wrong, but I checked all my components and they are all exported correctly with no typos.

Someone else said to use "render=" instead of "component=" but that is already happening as well:
https://stackoverflow.com/questions/37959896/warning-failed-proptype-invalid-prop-component-supplied-to-route

Has anyone encountered this and know how to address it? Thankfully it's not breaking anything.

**Comment by drteresavasquez** [comment]
Im looking at too many tickets at once 🤣. It's something with the PropTypes. Try changing component to any and see if it goes away.

**Comment by drteresavasquez** [comment]
@GonzalesMatthew Oh never mind. I see! Line 55 you have spaces between component = and the component. Get rid of those spaces.

**Comment by drteresavasquez** [comment]
@GonzalesMatthew please let us know if this cleared your issue and report on what the solution is.

**Comment by GonzalesMatthew** [comment, marked as answer]
I tried your fix and while that didn't make a difference it led me to the right answer ... my NotFound.js was empty! The routing was correct but I forgot to make a page when a user navigates to any other page not in my route tree. So if you see this error: (a) check your route exports, (b) check you even made a React component to begin with (my failure here 🤦

**Reply by drteresavasquez** [reply]
So if you have those spaces it didn't give you errors?

**Comment by GonzalesMatthew** [comment]
Yep! :)

**Reply by drteresavasquez** [reply]
That's interesting. I wouldn't format like that though. Make sure to remove those spaces so you don't look like a rookie 🤣
